### PR TITLE
rust : Issue #4597: fix lint warning about enum's name

### PR DESCRIPTION
**Pull request by hsadia538**
- [X] I have read the contributing guidelines at https://redmine.openinfosecfoundation.org/projects/suricata/wiki/Contributing
- [X] I have signed the Open Information Security Foundation contribution agreement at https://suricata.io/about/contribution-agreement/
- [X] I have updated the user guide (in doc/userguide/) to reflect the changes made (if applicable)

Link to [redmine](https://redmine.openinfosecfoundation.org/issues/4597) ticket:

Describe changes:
- Fixes Optimization issue #4597
- Fixes all lint warnings about "enum's name"
- Uncomments the check in lib.rs

### CODE REVIEW

File: doc/userguide/suricatasc.1 [hunk 1]
```diff
@@ -0,0 +1,244 @@
...
+.INDENT 2.0
+.INDENT 3.5
+\fI\%https://suricata.io/support/\fP
+.UNINDENT
+.UNINDENT
+.UNINDENT
+.SH COPYRIGHT
+2016-2022, OISF
+.\" Generated by docutils manpage writer.
+.
```
Comment (inashivb) [inline doc/userguide/suricatasc.1:244]: I don't think these man pages or your `.DS_Store` should be a part of the PR.

File: rust/src/lib.rs [hunk 2]
```diff
@@ -53,7 +53,7 @@
 #![allow(clippy::match_ref_pats)]
 #![allow(clippy::module_inception)]
 #![allow(clippy::needless_range_loop)]
-#![allow(clippy::enum_variant_names)]
+//#![allow(clippy::enum_variant_names)]
```
Comment (inashivb) [inline rust/src/lib.rs:56]: We should probably remove this instead of commenting it out.

**Review by inashivb — CHANGES_REQUESTED** [review]
Hello, @hsadia538!
Thank you for your first contribution to our project. :)
From a brief look, relevant changes to the relevant files look ok to me.

Following changes are requested by me:
1. the commit message title should indicate the part of code that is being changed. Here, since you're changing multiple files in the rust ecosystem, you could prefix it with `rust: `.
2. the commit message should have the reference to the ticket number. See commits in the existing PRs. e.g. https://github.com/OISF/suricata/pull/7949/commits/692a19469a38c24fcdbeda3dc78234a7622ef5ac
3. Any files not checked in the git tree of Suricata should not be a part of the PR. e.g. libhtp submodule, .DS_Store, etc

**Comment by hsadia538** [comment]
> Hello, @hsadia538! Thank you for your first contribution to our project. :) From a brief look, relevant changes to the relevant files look ok to me.
> 
> Following changes are requested by me:
> 
> 1. the commit message title should indicate the part of code that is being changed. Here, since you're changing multiple files in the rust ecosystem, you could prefix it with `rust: `.
> 2. the commit message should have the reference to the ticket number. See commits in the existing PRs. e.g. [692a194](https://github.com/OISF/suricata/commit/692a19469a38c24fcdbeda3dc78234a7622ef5ac)
> 3. Any files not checked in the git tree of Suricata should not be a part of the PR. e.g. libhtp submodule, .DS_Store, etc

Thank you for the detailed review, I'll close this PR and make a new one to address the changes mentioned. Thanks Alot once again!

File: doc/userguide/suricatasc.1 (hunk 1 above)
Comment (hsadia538) [reply doc/userguide/suricatasc.1:244]: Should I remove the userguide files (`.rst`) from the code as well?  I believe they are getting autogenerated from code.

Comment (inashivb) [reply doc/userguide/suricatasc.1:244]: Anything that is auto generated should be removed if it's not already a part of the tree, yes. Only the rust files that you change for the given issue should stay

Comment (hsadia538) [reply doc/userguide/suricatasc.1:244]: Alright, thank you for the clarification.

**Comment by hsadia538** [comment]
Followed by: #7988

**Pull request closed without merging**